### PR TITLE
idea for slight IntegrationManager refactor

### PR DESCRIPTION
**Pull request by MichaelRyanWebber**
to support loading Dynamics User Actions.
a more minimal change could be made by adding two calls to loadDynamicUserActions at the end of createIntegration and getInstanceFromIntegrationId and pushing the events. Happy to do that instead but wanted to jot this idea down for consideration

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @friggframework/api-module-activecampaign@0.8.26-canary.155.c43280b.0
  npm install @friggframework/api-module-airwallex@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-attentive@0.8.26-canary.155.c43280b.0
  npm install @friggframework/api-module-clyde@0.8.27-canary.155.c43280b.0
  npm install @friggframework/api-module-connectwise@0.8.28-canary.155.c43280b.0
  npm install @friggframework/api-module-crossbeam@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-fastspring-iq@0.8.27-canary.155.c43280b.0
  npm install @friggframework/api-module-front@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-gorgias@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-hubspot@0.8.28-canary.155.c43280b.0
  npm install @friggframework/api-module-huggg@0.8.26-canary.155.c43280b.0
  npm install @friggframework/api-module-ironclad@0.0.36-canary.155.c43280b.0
  npm install @friggframework/api-module-marketo@0.8.26-canary.155.c43280b.0
  npm install @friggframework/api-module-microsoft-teams@0.0.6-canary.155.c43280b.0
  npm install @friggframework/api-module-monday@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-netx@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-outreach@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-personio@0.8.26-canary.155.c43280b.0
  npm install @friggframework/api-module-pipedrive@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-qbo@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-rev-io@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-rollworks@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-salesforce@0.8.32-canary.155.c43280b.0
  npm install @friggframework/api-module-salesloft@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-slack@0.1.34-canary.155.c43280b.0
  npm install @friggframework/api-module-terminus@0.8.25-canary.155.c43280b.0
  npm install @friggframework/api-module-yotpo@0.0.19-canary.155.c43280b.0
  npm install @friggframework/api-module-zoom@0.8.25-canary.155.c43280b.0
  npm install @friggframework/core@0.2.13-canary.155.c43280b.0
  npm install @friggframework/integrations@1.1.1-canary.155.c43280b.0
  npm install @friggframework/module-plugin@1.0.26-canary.155.c43280b.0
  # or 
  yarn add @friggframework/api-module-activecampaign@0.8.26-canary.155.c43280b.0
  yarn add @friggframework/api-module-airwallex@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-attentive@0.8.26-canary.155.c43280b.0
  yarn add @friggframework/api-module-clyde@0.8.27-canary.155.c43280b.0
  yarn add @friggframework/api-module-connectwise@0.8.28-canary.155.c43280b.0
  yarn add @friggframework/api-module-crossbeam@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-fastspring-iq@0.8.27-canary.155.c43280b.0
  yarn add @friggframework/api-module-front@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-gorgias@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-hubspot@0.8.28-canary.155.c43280b.0
  yarn add @friggframework/api-module-huggg@0.8.26-canary.155.c43280b.0
  yarn add @friggframework/api-module-ironclad@0.0.36-canary.155.c43280b.0
  yarn add @friggframework/api-module-marketo@0.8.26-canary.155.c43280b.0
  yarn add @friggframework/api-module-microsoft-teams@0.0.6-canary.155.c43280b.0
  yarn add @friggframework/api-module-monday@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-netx@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-outreach@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-personio@0.8.26-canary.155.c43280b.0
  yarn add @friggframework/api-module-pipedrive@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-qbo@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-rev-io@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-rollworks@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-salesforce@0.8.32-canary.155.c43280b.0
  yarn add @friggframework/api-module-salesloft@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-slack@0.1.34-canary.155.c43280b.0
  yarn add @friggframework/api-module-terminus@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/api-module-yotpo@0.0.19-canary.155.c43280b.0
  yarn add @friggframework/api-module-zoom@0.8.25-canary.155.c43280b.0
  yarn add @friggframework/core@0.2.13-canary.155.c43280b.0
  yarn add @friggframework/integrations@1.1.1-canary.155.c43280b.0
  yarn add @friggframework/module-plugin@1.0.26-canary.155.c43280b.0
  ```
</details>

### CODE REVIEW

**Comment by MichaelRyanWebber** [comment]
also thinking we could add getUserActions() to the integration returns an empty array by default, so that loadDynamicActions wouldn't need to be overridden and the Integration could just override getUserActions if needed

File: packages/integrations/manager.js [hunk 1]
```diff
@@ -57,23 +57,9 @@ class IntegrationManager extends Delegate {
 
         const instance = await integrationManagerClass.getInstance({
             userId,
-            integrationId: params.integrationId,
+            events: integrationManagerClass.Config.events
```
Comment (leofmds) [inline packages/integrations/manager.js:60]: Is there any chance this will impact previous implementations?

**Review by leofmds — COMMENTED** [review]
Well done @MichaelRyanWebber
I liked this approach, it certainly facilitates a lot for users to create their own integrations.

File: packages/integrations/manager.js (hunk 1 above)
Comment (MichaelRyanWebber) [reply packages/integrations/manager.js:60]: @MichaelRyanWebber  thanks for the review. Just for reference this PR is pretty old and doesn't indicate the bigger refactor I brought up in [slack](https://lefthookhq.slack.com/archives/C017RNCQV70/p1687460445053899)

This shouldn't cause a problem for previous implementations, even though it looks like it. While getInstance was passed the integrationId, it didn't actually do anything with it. It passed down into the Delegate constructor and ignored. 

In terms of the events, while this change means getInstance now supports supplying events on instantiation, it does not require it and reverts to the previous behavior if they are absent. 

That said, this was all rather confusing to begin with, and could use more work that could stray into breaking changes. The existing Frigg Apps i've seen tend to override the `getInstanceFromIntegrationId` but continue to use the base class implementation of `getInstance`, so I held that as a limitation for now.

Comment (MichaelRyanWebber) [reply packages/integrations/manager.js:60]: I do think the change to getInstance and the Delegate constructor could be problematic. Technically the previous iteration instantiated two `IntegrationManager`s with the first as the second's Delegate. The first gets supplied with events (i.e. delegateTypes) but the second does not as part of `getInstance`.

The approach I switched to makes the `delegate` point back to itself by default. But there's a possibility I misunderstand the intentions or all the use-cases involved. Thoughts welcome!

**Pull request closed without merging**